### PR TITLE
[Typing][A-81] Add type annotations for `paddle/vision/models/resnet.py`

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR Category

User Experience
### PR Types

Improvements

### Description

Add type annotations for `paddle/vision/models/resnet.py`
- #65008

### CODE REVIEW

File: python/paddle/vision/models/resnet.py [hunk 1]
```diff
@@ -75,22 +107,22 @@
...
-        norm_layer=None,
-    ):
+        inplanes: int,
+        planes: int,
+        stride: Size2 = 1,
+        downsample: nn.Sequential | None = None,
+        groups: int = 1,
+        base_width: int = 64,
+        dilation: int = 1,
+        norm_layer: type[nn.BatchNorm2D] | None = None,
```
Comment (SigureMo) [inline python/paddle/vision/models/resnet.py]: `type[nn.Layer]`？这要是固定 `nn.BatchNorm2D` 就没意义了吧？

还是说意义就是用来区分 `None` 和 `BatchNorm2D`？

File: python/paddle/vision/models/resnet.py [hunk 2]
```diff
@@ -75,22 +107,22 @@
...
-        downsample=None,
-        groups=1,
-        base_width=64,
-        dilation=1,
-        norm_layer=None,
-    ):
+        inplanes: int,
+        planes: int,
+        stride: Size2 = 1,
+        downsample: nn.Sequential | None = None,
```
Comment (SigureMo) [inline python/paddle/vision/models/resnet.py]: 这个可以是其他 Layer 吗？

File: python/paddle/vision/models/resnet.py [hunk 3]
```diff
@@ -12,13 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import TYPE_CHECKING, Unpack
```
Comment (megemini) [inline python/paddle/vision/models/resnet.py]: from typing_extensions import Unpack

File: python/paddle/vision/models/resnet.py (hunk 1 above)
Comment (Asthestarsfalll) [reply python/paddle/vision/models/resnet.py]: 看代码ResNet里是写死为BatchNorm2D了，使用Layer也可以

File: python/paddle/vision/models/resnet.py (hunk 3 above)
Comment (Asthestarsfalll) [reply python/paddle/vision/models/resnet.py]: 稍后修改

File: python/paddle/vision/models/resnet.py (hunk 2 above)
Comment (Asthestarsfalll) [reply python/paddle/vision/models/resnet.py]: 也可以 稍后修改

**Review by SigureMo — DISMISSED** [review]
LGTMeow

**Comment by SigureMo** [comment]
啊不对，Static check 有点问题得看看（刚刚手快直接 rerun 了

File: python/paddle/vision/models/resnet.py
Comment (megemini) [inline python/paddle/vision/models/resnet.py:1]: 导入的时候注意一下：

``` python
if TYPE_CHECKING:
    from typing import Literal, TypedDict
    from typing_extensions import Unpack, NotRequired
```

最好本地用 python 3.8 ~ 不然有些问题可能察觉不到 ～

Comment (Asthestarsfalll) [reply python/paddle/vision/models/resnet.py:1]: okk

**Review by SigureMo — APPROVED** [review]
